pkg/plugin/loader/loader.go

```diff
@@ -53,98 +50,139 @@ func LoadAllPlugins(namespace string, searchPath []string, selections []plugin.S
+			return nil, errors.Wrapf(err, "couldn't load plugin %v", def.SonobuoyConfig.PluginName)
+		}
+		plugins = append(plugins, loadedPlugin)
 	}
-}
 
-// scanPlugins looks for Plugin Definition files in the given directory,
-// and returns an array of PluginDefinition structs.
-func scanPlugins(dir string) ([]plugin.Definition, error) {
-	var pluginDfns []plugin.Definition
+	return plugins, nil
+}
 
-	files, err := ioutil.ReadDir(dir)
+func findPlugins(dir string) ([]string, error) {
+	candidates, err := ioutil.ReadDir(dir)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to read plugin directory")
+		return []string{}, errors.Wrapf(err, "couldn't search path %v", dir)
 	}
 
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".tmpl" {
-			logrus.WithField("filename", file.Name()).Info("unknown template type")
+	plugins := []string{}
+	for _, candidate := range candidates {
+		if candidate.IsDir() {
 			continue
 		}
-
-		// Read the template file into memory
-		fullPath := path.Join(dir, file.Name())
-		pluginTemplate, err := ioutil.ReadFile(fullPath)
-		if err != nil {
-			return nil, err
-		}
-		dfn, err := loadTemplate(pluginTemplate)
-		if err != nil {
-			logrus.WithError(err).WithField("filename", file.Name()).Info("failed to load plugin")
-			continue
+		ext := filepath.Ext(candidate.Name())
+		if ext == ".yml" || ext == ".yaml" {
+			plugins = append(plugins, filepath.Join(dir, candidate.Name()))
 		}
-		pluginDfns = append(pluginDfns, *dfn)
 	}
+	return plugins, nil
+}
+
+type sonobuoyConfig struct {
+	Driver     string `json:"driver"`
+	PluginName string `json:"plugin-name"`
+	ResultType string `json:"result-type"`
+}
 
-	return pluginDfns, err
+type pluginDefinition struct {
+	SonobuoyConfig sonobuoyConfig   `json:"sonobuoy-config"`
+	Spec           corev1.Container `json:"spec"`
 }
 
-func loadTemplate(tmpl []byte) (*plugin.Definition, error) {
-	t, err := template.New("plugin").Parse(string(tmpl))
+func loadDefinition(file string) (*pluginDefinition, error) {
+	bytes, err := ioutil.ReadFile(file)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse template")
+		return nil, errors.Wrapf(err, "couldn't open lugin definition %v", file)
+	}
+
+	// convert to JSON because corev1.Container only has JSON tags
+	var decoded interface{}
+	if err = yaml.Unmarshal(bytes, &decoded); err != nil {
+		return nil, errors.Wrapf(err, "couldn't decode yaml for plugin definition %v", file)
 	}
-	var b bytes.Buffer
-	// We just trying to get a kubernetes object here we don't really care about values rn
-	err = t.Execute(&b, &plugin.DefinitionTemplateData{})
+
+	decoded = convert(decoded)
+
+	jsonBytes, err := json.Marshal(decoded)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to execute template")
+		return nil, errors.Wrapf(err, "couldn't encode yaml as json for plugin definition %v", file)
+	}
+
+	var internalDef pluginDefinition
+	if err = json.Unmarshal(jsonBytes, &internalDef); err != nil {
```

Comment:
Definitely document the logic behind the decode + convert + marshal + unmarshal ... hopefully we can find our way around that later.

Reply:
@timothysc lucky you, my next PR does away with this altogether!